src/app/server/CommissioningWindowManager.cpp

```diff
@@ -162,6 +166,10 @@ void CommissioningWindowManager::OnSessionEstablishmentStarted()
     // As per specifications, section 5.5: Commissioning Flows
     constexpr System::Clock::Timeout kPASESessionEstablishmentTimeout = System::Clock::Seconds16(60);
     DeviceLayer::SystemLayer().StartTimer(kPASESessionEstablishmentTimeout, HandleSessionEstablishmentTimeout, this);
+    if (mAppDelegate != nullptr)
+    {
+        mAppDelegate->OnCommissioningSessionEstablishmentStarted();
```

Comment:
OK.  So just to be clear, given this setup we can have a `OnCommissioningSessionEstablishmentFailed` without a preceding `OnCommissioningSessionEstablishmentStarted`.  I that expected behavior for the appdelegate API here?

Reply:
These are unsolicited events and hence it is expected.
But,if the events are following the order, there is less probability that they will occur out of sync.

Reply:
> These are unsolicited events and hence it is expected.

No, I mean by the API consumer.  Is it clear from reading the API that OnCommissioningSessionEstablishmentFailed can happen without a corresponding OnCommissioningSessionEstablishmentStarted?

>  But,if the events are following the order, there is less probability that they will occur out of sync.

I don't understand this sentence.